test/gtest/intervals.cpp

```diff
@@ -0,0 +1,31 @@
+#include "support/intervals.h"
+#include "ir/utils.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+TEST(IntervalsTest, TestNoOverlapFound) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 4, 2});
+  intervals.emplace_back(Interval{4, 8, 2});
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals).size(), 2u);
```

Comment:
It would be better to test the precise results rather than just the size of the results. You can still do it with a single ASSERT_EQ:

```
std::vector<int> expected{0, 1};
ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
```

Reply:
done